### PR TITLE
Integrate each sample over weighted offsets in `PointingOperator`

### PR DESCRIPTION
**Pull request by ArtemBasyrov**
## What

A detector can now read the sky at several directions around its pointing and return their weighted sum. `PointingOperator.create` and `XSamplingOperator.create` take `offsets` (detector-frame quaternions, shared or per detector) and `offset_weights` (one per offset, or one row per Stokes component). With `offsets=None` every code path is unchanged.

This models an integration within a sample: a finite time integration, a pixel window, or a beam. The beam is the target use, with the nodes and weights of a gridded beam to come in a follow-up.

## How

Each offset direction gets its own stencil, and `Stencil.integrated` folds the offset axis into the neighbour axis with the offset weights multiplied in. `_sample` and `_bin` consume that wide stencil as they would a bilinear one. The transport target stays the un-offset direction, so the Q and U of every pixel read, whichever offset reads it, are carried into one frame before the sum. The stencil weights may carry a leading Stokes axis, so a beam can weight I, Q and U differently.

`transported_scatter` now applies the stencil weights before rotating back into the pixel frames. This is what keeps it adjoint to `transported_gather` when the weights differ between Q and U. With equal weights the result changes at round-off level only.

## Notes

- `as_stokes_i` keeps the offsets and reduces per-Stokes weights to the I row, so the hitmap is unchanged.
- `as_expanded_operator` carries the offsets through `XSamplingOperator`, whose angle cache then grows by `n_offsets`. A chunked offset axis is left to a follow-up.
- `AtmospherePointingOperator` rejects offsets: its wind displacement is added per sample and does not line up with an offset axis.
- Offset weights are normalised to sum to one over the offsets in the map, the same convention as a partly covered bilinear sample.

## Tests

`TestOffsets` in `tests/obs/test_pointing.py` covers both frames and both interpolations: the origin offset with unit weight equals the plain operator, two offsets at equal weight equal the mean of two detectors carrying them, per-detector offsets match shared ones, the adjoint holds with per-Stokes weights, and the expanded operator matches the on-the-fly one. `TestIntegrated` in `tests/obs/test_stencil.py` covers the fold, the off-map renormalisation and per-Stokes rows. `tests/obs/spin2/test_sampling.py` checks the scatter is the transpose of the gather with a weight per component.

### CODE REVIEW

File: src/furax/obs/pointing.py [hunk 1]
```diff
@@ -73,6 +84,9 @@ class PointingOperator(AbstractLinearOperator):
     batch_size: int = field(metadata={'static': True})
     interpolate: bool = field(metadata={'static': True})
     _out_structure: PyTree[jax.ShapeDtypeStruct] = field(metadata={'static': True})
+    # keyword-only so that subclasses can still declare required fields
+    offsets: Quaternion | None = field(default=None, kw_only=True)
+    offset_weights: OffsetWeights | None = field(default=None, kw_only=True)
```
Comment (sbiquard) [inline src/furax/obs/pointing.py]: Should this be `Array | Stokes | None` so we rely less on counting rows? The `stokes` attribute of the weights would have to match the landscape's. I guess it can stay generic at the stencil level.

File: src/furax/obs/pointing.py [hunk 2]
```diff
@@ -261,6 +347,27 @@ def _quat2pointing(self, qdet_full: Quaternion) -> SampledPointing:
...
+            qdet_full: The pointing of every sample, shape (det, samp).
+            offsets: The offsets of those detectors, shape (det, n_offsets), or `None`.
+        """
+        if offsets is None:
+            return self._quat2pointing(qdet_full)
+        assert self.offset_weights is not None
+        theta, phi = self.landscape.quat2world(qdet_full)
+        # (det, samp, 1) x (det, 1, n_offsets) -> (det, samp, n_offsets), one direction per offset
+        offset_pointing = self._quat2pointing(qdet_full[:, :, None] * offsets[:, None, :])
+        stencil = offset_pointing.stencil.integrated(self.offset_weights)
```
Comment (sbiquard) [inline src/furax/obs/pointing.py]: As far as I can tell, these offset_weights are applied in the sky frame. Does this mean that the Q and U weights depend on the coordinate system and/or parallactic angle? Would it not make more sense for these weights to apply in the detector frame?

**Review by sbiquard — CHANGES_REQUESTED** [review]

File: src/furax/obs/pointing.py (hunk 1 above)
Comment (ArtemBasyrov) [reply src/furax/obs/pointing.py]: agreed, and since weights should include an option for different weights for different Stokes component.

File: src/furax/obs/pointing.py (hunk 2 above)
Comment (ArtemBasyrov) [reply src/furax/obs/pointing.py]: they need to be applied in the same frame, otherwise they cannot be summed over. but you're right and I fixed that frame is rotated to local detector frame first

**Review by sbiquard — COMMENTED** [review]
I think the changelog lines could be tightened, but the PointingOperator redesign will make them obsolete anyway, so let's leave it like that for now

**Review by sbiquard — APPROVED** [review]
